File: sample/java-groovy-dsl/src/main/java/io/github/simonschiller/prefiller/sample/customer/CustomerDao.java
package io.github.simonschiller.prefiller.sample.customer;

import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

@Dao
public interface CustomerDao {

    @Query("SELECT * FROM customers")
    List<Customer> getAll();

    @Query("SELECT * FROM adults")
    List<AdultCustomer> getAdults();
}

File: sample/kotlin-ksp/src/main/kotlin/io/github/simonschiller/prefiller/sample/customer/CustomerDao.kt
package io.github.simonschiller.prefiller.sample.customer

import androidx.room.Dao
import androidx.room.Query

@Dao
interface CustomerDao {

    @Query("SELECT * FROM customers")
    fun getAll(): List<Customer>

    @Query("SELECT * FROM adults")
    fun getAdults(): List<AdultCustomer>
}
